R/as_OpenSpecy.R

```diff
@@ -260,8 +260,16 @@ as_OpenSpecy.default <- function(x, spectra,
   obj <- structure(list(), class = c("list", "OpenSpecy"))
 
   obj$wavenumber <- x
+  
+  if(!(identical(order(x), 1:length(x)) | identical(order(x), length(x):1))){
+      warning("Wavenumbers should be a continuous sequence for all OpenSpecy functions to run smoothly. Consider sorting your wavenumbers and spectral intensities before turning into an OpenSpecy object.")
+  }
+  
   obj$spectra <- as.data.table(spectra)
 
+  if(length(unique(colnames(obj$spectra))) != ncol(obj$spectra)){
+      warning("Column names should be unique in the spectral intensities otherwise some functions may not work. Consider creating new names for the spectral data or collapsing spectra with the same name.")
+  }
```

Comment:
Duplicate column names could give us a lot of trouble and are rather bad practice. I'd suggest to turn this warning into an error here.

Reply:
Good point, updated. https://github.com/wincowgerDEV/OpenSpecy-package/pull/129/commits/6a66b43263314880d066d423934d62f2d999141d